src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFChart.java

```diff
@@ -105,19 +136,218 @@ public boolean equals(Object obj) {
         if (obj == this) {
             return true;
         }
-
+        
         if (obj == null) {
             return false;
         }
-
+        
         if (!(obj instanceof XWPFChart)) {
             return false;
         }
         return false;
     }
-
+    
     @Override
     public int hashCode() {
         return getChecksum().hashCode();
     }
+    
+    /**
+     * method to create relationship with embedded part for example writing xlsx
+     * file stream into output stream
+     * 
+     * @param chartRelation chart relationship object which used to create relation with chart
+     * @param index its chart index
+     * @return return relation part which used to write relation in .rels file
+     *         and get relation id
+     * @since POI 4.0.0
+     */
+    public RelationPart createRelationship(XWPFRelation chartRelation, int index) {
+        return createRelationship(chartRelation, XWPFFactory.getInstance(), index, false);
+    }
+    
+    /**
+     * protected method which used to initialization of sheet output stream
+     * 
+     * @param sheet output stream of embedded xlsx file
+     * @since POI 4.0.0
+     */
+    protected void addEmbeddedWorkSheet(OutputStream sheet) {
+        this.sheet = sheet;
+    }
+    
+    /**
+     * this method is used to write workbook object in embedded part of chart
+     * return true in case of successfully write work book in embedded part or
+     * return false
+     * 
+     * @param wb this is workbook object which we used to write in embedded xlsx file
+     * @return returns true in case of successfully write workbook in embedded
+     *         part or returns false
+     * @throws IOException
+     * @since POI 4.0.0
+     */
+    public boolean writeEmbeddedWorkSheet(Workbook wb) throws IOException {
+        if (this.sheet != null && wb != null) {
+            wb.write(this.sheet);
+            return true;
+        }
+        return false;
+    }
+    
+    /**
+     * initialize in line object
+     * 
+     * @param inline this object is used to adjust the margin and dimension of chart 
+     * @since POI 4.0.0
+     */
+    protected void setInLine(CTInline ctInline) {
```

Comment:
All methods using `CT*` classes should be annotated with `@Internal` to clearly state their purpose is not for general public use.

Reply:
updated suggested changes